### ISSUE TITLE
Document the .ignore format

### ISSUE THREAD

**Issue by petdance** [ggreer/the_silver_searcher#1171]
I don't see any documentation of what can go into the .ignore file.  Single file patterns?  Full pathnames?  Are they all globs?  Regexes?

**Comment by bgold09** [comment]
I believe this is covered in the [wiki](https://github.com/ggreer/the_silver_searcher/wiki/Advanced-Usage#ignore), specifically: 

> `.ignore` uses the same patterns as `.gitignore` and `.hgignore`.

**Comment by petdance** [comment]
I suggest that having to go read a wiki, and having to know that you should go to the wiki, is not very useful as documentation.

**Comment by mcdemarco** [comment]
It's not just having to know to go read a wiki, but then having to go look up the gitignore format, and then realizing there are still bugs in ag related to not supporting the full gitignore format, and then reading those bugs to try to figure out what actually works in ag.  The last couple of comments on #385 are helpful in that regard, but actual documentation would be more helpful.

**Comment by lelutin** [comment]
This and the fact that the man page does not mention where to find more information (... and actually I hate man pages that point to a web site, it's always better to have documentation right there when you ask for it)

The man page says nothing about what consitutes a PATTERN and that's really puzzling for users

**Comment by BurntSushi** [comment]
@lelutin For pattern syntax, see `man pcresyntax`.

**Comment by petdance** [comment]
> For pattern syntax, see `man pcresyntax`

I'd say that would be a good thing to have in there then.

**Comment by lelutin** [comment]
@petdance indeed, the whole point of this ticket is that the manpage mentions nowhere what constitutes a pattern. a short sentence with a reference to another manpage could be enough to remove abiguities

**Comment by zeorin** [comment]
It would also be useful to know beforehand whether comments are supported (seems like they are).
